src/main/java/io/rainfall/ScenarioRun.java

```diff
@@ -131,21 +137,45 @@ public ScenarioRun assertion(final Assertion actual, final Assertion expected) {
         reportingConfig.getStatisticsCollectors());
     initStatistics(this.statisticsHolder);
 
-    ScheduledExecutorService topOfSecondExecutor = Executors.newSingleThreadScheduledExecutor();
+    List<ScheduledExecutorService> topOfSecondExecutors = new ArrayList<>();
```

Comment:
```
ScheduledExecutorService topOfSecondExecutors = Executors.newScheduledThreadPool(reportingConfig.getLogReporters().size())
```

seems good enough, even if you wanted to schedule only one log reporter you could use just one thread from the pool

Reply:
So when a user doesn't specify a reporter-specific interval, we'd end up creating a thread pool with multiple threads, but will end up using just one out of them. This might have been okay if this were just a fixed, one-time job. Given that this job will run periodically, we'd end up occupying resources unnecessarily for a significant period of time.

Reply:
Done after offline discussion.